strax/plugin.py

```diff
@@ -648,18 +648,35 @@ def compute(self, **kwargs):
                                        f"should be {len(base)}!")
                 kwargs[k] = r
 
-        results = np.zeros(len(base), dtype=self.dtype)
-        deps_by_kind = self.dependencies_by_kind()
-
-        for i in range(len(base)):
-            r = self.compute_loop(base[i],
-                                  **{k: kwargs[k][i]
-                                     for k in deps_by_kind
-                                     if k != loop_over})
-
-            # Convert from dict to array row:
-            for k, v in r.items():
-                results[i][k] = v
+        if not self.multi_output:
```

Comment:
Minor comment can we revert the logic, `if self.multi_output:` is a nano bit simpler.

Reply:
Sure, if it makes you but a nano bit more happy, I'm completely fine with it. Just though most people read from top to bottom and no one has been having `self.multi_output` issues .